x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractAggregationTestCase.java

```diff
@@ -155,6 +164,63 @@ private void aggregateSingleMode(Expression expression) {
         }
     }
 
+    private void aggregateGroupingSingleMode(Expression expression) {
+        var pages = rows(testCase.getMultiRowFields());
+        List<Object> results;
+        try {
+            assumeFalse("Grouping aggregations must receive data to check results", pages.isEmpty());
```

Comment:
You should be able to check what happens with an empty set of `SeenGroupIds`

Reply:
I could call `prepareProcessPage()` with the empty SeenGroupIds, but then I would call `evaluateFinal()` with an empty IntVector of selected groups, which would do nothing. So I'm not sure if it makes sense to test this (?)

Reply:
We do sometimes send no values to aggs. `ROW a = 1 | WHERE a < 0 | STATS MIN(a) BY a` would do it. It's a valid thing in production so if it's not a huge rework I think it'd be nice to test it.